--- a/configure.ac	2013-04-19 12:21:27.328079635 -0400
+++ b/configure.ac	2013-04-19 12:21:46.363079979 -0400
@@ -843,16 +843,12 @@
 
         if test "$with_gui" != no; then
 
-                if test $XCOS_ENABLE = yes; then
-                   # jgraphx
-                   AC_JAVA_CHECK_PACKAGE([jgraphx],[com.mxgraph.model.mxCell],[Diagram design])
-                   JGRAPHX=$PACKAGE_JAR_FILE
-                   AC_SUBST(JGRAPHX)
-
-                   AC_JAVA_CHECK_VERSION_PACKAGE([jgraphx],[import com.mxgraph.view.mxGraph;],$JGRAPHX,[1.8.0.0],[mxGraph.VERSION],[],[])
-
-                fi
+                # jgraphx
+                AC_JAVA_CHECK_PACKAGE([jgraphx],[com.mxgraph.model.mxCell],[Diagram design])
+                JGRAPHX=$PACKAGE_JAR_FILE
+                AC_SUBST(JGRAPHX)
 
+                AC_JAVA_CHECK_VERSION_PACKAGE([jgraphx],[import com.mxgraph.view.mxGraph;],$JGRAPHX,[1.8.0.0],[mxGraph.VERSION],[],[])
 
                 # scirenderer
                 AC_JAVA_CHECK_PACKAGE([scirenderer],[org.scilab.forge.scirenderer.PackageInfo],[Scilab Renderer])
